src/univers/version_constraint.py

```diff
@@ -0,0 +1,221 @@
+        Return a tuple of (comparator, version) strings given a
+        constraint ``string`` such as ">=2.3".
+
+        For example::
+        >>> assert VersionConstraint.split(">=2.3") == (">=", "2.3",)
+        >>> assert VersionConstraint.split("  <   =  2 . 3  ") == ("<=", "2.3",)
+        >>> assert VersionConstraint.split("2.3") == ("=", "2.3",)
+        >>> assert VersionConstraint.split("*2.3") == ("*", "",)
+        >>> assert VersionConstraint.split("*") == ("*", "",)
+        >>> assert VersionConstraint.split("<2.3") == ("<", "2.3",)
+        >>> assert VersionConstraint.split(">2.3") == (">", "2.3",)
+        >>> assert VersionConstraint.split("!=2.3") == ("!=", "2.3",)
+        """
+        constraint_string = remove_spaces(string)
+
+        # special case for star
+        if constraint_string.startswith("*"):
+            return "*", ""
+
+        for comparator in COMPARATORS:
+            if constraint_string.startswith(comparator):
+                # we do not report an error if this is not valid
+                version = constraint_string.lstrip("><=!")
+                if comparator == "*":
+                    version = ""
+                return comparator, version
+
+        # default to equality
+        return "=", constraint_string
+
+    # FIXME: this may be not enough to only handle "contains"?
+    def __contains__(self, version):
+        """
+        Return a True if the ``version`` Version is contained in this
+        VersionConstraint or "satisfies" this VersionConstraint.
+
+        For example::
+        >>> from univers.versions import PypiVersion
+        >>> v22 = PypiVersion("2.2")
+        >>> v23 = PypiVersion("2.3")
+        >>> v24 = PypiVersion("2.4")
+        >>> assert v23 in VersionConstraint(comparator="=", version=v23)
+        >>> assert v24 not in VersionConstraint(comparator="=", version=v23)
+        >>> assert None not in VersionConstraint(comparator="=", version=v23)
+
+        >>> assert None not in VersionConstraint(comparator="!=", version=v24)
+        >>> assert v22 in VersionConstraint(comparator="!=", version=v23)
+        >>> assert v23 in VersionConstraint(comparator="!=", version=v24)
+        >>> assert v24 not in VersionConstraint(comparator="!=", version=v24)
+
+        >>> assert v24 in VersionConstraint(comparator=">", version=v23)
+        >>> assert v23 not in VersionConstraint(comparator=">", version=v23)
+        >>> assert v24 in VersionConstraint(comparator=">=", version=v23)
+        >>> assert v23 in VersionConstraint(comparator=">=", version=v23)
+        >>> assert v22 not in VersionConstraint(comparator=">=", version=v23)
+
+        >>> assert v23 in VersionConstraint(comparator="<", version=v24)
+        >>> assert v23 in VersionConstraint(comparator="<=", version=v24)
+        >>> assert v24 in VersionConstraint(comparator="<=", version=v24)
+        >>> assert v24 not in VersionConstraint(comparator="<", version=v24)
+        """
+        if not version:
```

Comment:
But for "*" comparator, version is supposed to be None.  Hence it will always return false which is opposite of what we're expecting.

Reply:
The `version` argument here is the version that we want to check. This is not the same as the version attribute of the constraint. You would never pass a constraint as an argument. The `if not version:` is just a guard against calling the method using `None` and other falsish values... but to your point this can be confusing and would be caught by the type check done below... so I am removing this line.